hbook/h1d.go

```diff
@@ -186,6 +186,105 @@ func (h *H1D) Scale(factor float64) {
 	h.Binning.scaleW(factor)
 }
 
+// AddH1D returns the bin-by-bin summed histogram of h1 and h2
+// assuming their statistical uncertainties are uncorrelated.
+func AddH1D(h1, h2 *H1D) *H1D {
+
+	if h1.Len() != h2.Len() {
+		panic("hbook: h1 and h2 have different number of bins")
+	}
+
+	if h1.XMin() != h2.XMin() {
+		panic("hbook: h1 and h2 have different Xmin")
+	}
+
+	if h1.XMax() != h2.XMax() {
+		panic("hbook: h1 and h2 have different Xmax")
+	}
+
+	hsum := NewH1D(h1.Len(), h1.XMin(), h1.XMax())
+
+	for i := 0; i < hsum.Len(); i++ {
+		y := h1.Value(i) + h2.Value(i)
+		y1err2 := h1.Binning.Bins[i].SumW2()
+		y2err2 := h2.Binning.Bins[i].SumW2()
+		hsum.Binning.Bins[i].Dist.Dist.SumW = y
+		hsum.Binning.Bins[i].Dist.Dist.SumW2 = y1err2 + y2err2
+	}
+
+	// handle under/over flow
+	for i := range hsum.Binning.Outflows {
+		y := h1.Binning.Outflows[i].Dist.SumW + h2.Binning.Outflows[i].Dist.SumW
+		y1err2 := h1.Binning.Outflows[i].Dist.SumW2
+		y2err2 := h2.Binning.Outflows[i].Dist.SumW2
+		hsum.Binning.Outflows[i].Dist.SumW = y
+		hsum.Binning.Outflows[i].Dist.SumW2 = y1err2 + y2err2
+	}
+
+	return hsum
+}
+
+// AddScaledH1D returns the histogram with the bin-by-bin h1+alpha*h2
+// operation, assuming statistical uncertainties are uncorrelated.
+func AddScaledH1D(h1 *H1D, alpha float64, h2 *H1D) *H1D {
+	h2.Scale(alpha)
```

Comment:
this modifies `h2` in place.

perhaps we should turn this around and implement `AddH1D` in terms of `AddScaledH1D(h1, 1, h2)` instead?